File: Day02/Lab03.js
let numbers = [1, 2, 3, 4, 5, 6, 30, 56, 84, 23, 100];
/*
Print all the numbers
Print the sum of all numbers
Print the first odd number
Generate an list of squares of all numbers
Generate a list of numbers divisible by 5.
Print all the even numbers

*/

numbers.forEach(it => console.log(it));

const sum = numbers.reduce((prevValue, currentValue) => prevValue + currentValue);
console.log("Sum", sum);

const firstOddNumber = numbers.find(num => num % 2 !== 0);
console.log("First odd number", firstOddNumber);

const lstOfSquares = numbers.map(num => num * num);
console.log("List of squares of all numbers", lstOfSquares);

const lstOfNumbersDivBy5 = numbers.filter(it => it % 5 === 0);
console.log("List of numbers divisible by 5", lstOfNumbersDivBy5);
   
const lstEvenNumbers = numbers.filter(it => it % 2 === 0);
console.log("List of even numbers", lstEvenNumbers);

File: Day02/Lab03.java
import java.util.Arrays;
import java.util.List;
import java.util.stream.Collectors;

public class Lab03 {
    public static void main(String[] args) {
        List<Integer> numbers = Arrays.asList(1, 2, 3, 4, 5, 6, 30, 56, 84, 23, 100);
        
        numbers.forEach(it -> System.out.println(it));

        final int sum = numbers
            .stream()
            .reduce((prevValue, currentValue) -> prevValue + currentValue)
            .get();
        System.out.println("Sum: " + sum);
        
        final int firstOddNumber = numbers
            .stream()
            .filter(it -> it % 2 != 0)
            .findFirst()
            .get();
        System.out.println("First odd number: " + firstOddNumber);
        
        final List<Integer> lstOfSquares = numbers
            .stream()
            .map(it -> it * it)
            .collect(Collectors.toList());

        System.out.println("List of squares: " + lstOfSquares);
     
        
    }
}    